fuel-asm/src/lib.rs

```diff
@@ -308,6 +308,8 @@ impl_instructions! {
     0x97 POPL popl [bitmask: Imm24]
     "Pop a bitmask-selected set of registers in range 40..64 to the stack."
     0x98 POPH poph [bitmask: Imm24]
+    "Store return address and jump to an absolute address."
+    0x99 JAL jal [ret_addr: RegId target: RegId offset: Imm12]
```

Comment:
We might want something like:

```
0x99 JAL jal [ret_addr: RegId target: Imm12]
0x9A JALR jalr [ret_addr: RegId target: RegId offset: Imm12]
```

Reply:
I don't see much reason to do this. The form above with only immediate wouldn't be too useful, since you can already do that with setting target register to `Reg::ZERO`. Having a wider immediate might help in some huge programs, but only a little bit.

Reply:
makes sense, seems like a very minor/niche optimization